packages/kmod-5.15-nvidia/disable-gsp.service.in

```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=Disable GSP for selected instance types
+RequiresMountsFor=/etc/modprobe.d PREFIX/share/factory/etc/modprobe.d
+After=network-online.target
+Before=load-kernel-modules.service
+# Rerunning this service after the system is fully loaded will accomplish
+# nothing except write to a file that will never be used (after boot).
+# Refuse manual intervention.
+RefuseManualStart=true
+RefuseManualStop=true
+
+[Service]
+Type=oneshot
+ExecCondition=/usr/bin/shibaken does-value-start-with instance-type \
```

Comment:
One drawback is that because we transition targets several times (to configured, then to multi-user) this check will be repeated on other instance types (since the service won't have started and won't still be running).

You can avoid that by creating and checking for a sentinel file:
```
[Unit]
...
ConditionPathExists=!/etc/.gsp-disable-check-ran

[Service]
Type=oneshot
# This always returns true, but creates the sentinel file to prevent future checks.
ExecCondition=/usr/bin/touch /etc/.gsp-disable-check-ran
# Only proceed on specific instance types.
ExecCondition=/usr/bin/shibaken does-value-start-with instance-type \
  --value g4dn. --value g5. --value g5g.
...
```

Reply:
I can do that. The actual state changing cp execution is idempotent, but the IMDS query is not free, so let's not repeat it.